dart/constraint/SmartPointer.hpp

```diff
@@ -51,6 +51,8 @@ DART_COMMON_MAKE_SHARED_WEAK(LCPSolver)
 
 DART_COMMON_MAKE_SHARED_WEAK(BallJointConstraint)
 DART_COMMON_MAKE_SHARED_WEAK(WeldJointConstraint)
+DART_COMMON_MAKE_SHARED_WEAK(HumanArmJointLimitConstraint)
```

Comment:
Since these types are only defined in example code, I think they should go in a header of that example code instead of a public DART header.